cve_bin_tool/checkers/gcc.py

```diff
@@ -12,10 +12,25 @@
 
 
 class GccChecker(Checker):
-    CONTAINS_PATTERNS = []
+    # NOTE: find version string for debian packages
+    CONTAINS_PATTERNS = [
+        r"Do not predefine system-specific and GCC-specific macros\.",
+        r"Dump detailed information on GCC's internal representation of source code locations\.",
+        # r"GCC is not configured to support %s as offload target",
+        # r"IPA lattices after constant propagation, before gcc_unreachable:",
+        # r"Record gcc command line switches in DWARF DW_AT_producer\.",
+        # r"Record gcc command line switches in the object file\.",
+        # r"Warn about packed bit-fields whose offset changed in GCC [0-9]\.[0-9]\.",
+        # r"binary constants are a C\+\+14 feature or GCC extension",
+        # r"fixed-point constants are a GCC extension",
+        # r"gcc driver version %s %sexecuting gcc version %s",
+        # r"offset of packed bit-field %qD has changed in GCC [0-9]\.[0-9]",
+        # r"style of line directive is a GCC extension",
+        # r"suffix for double constant is a GCC extension",
+    ]
     FILENAME_PATTTERN = [r"gcc"]
     VERSION_PATTERNS = [
-        r"gcc ([0-9]+\.[0-9]+\.[0-9]+)",
-        r"gcc ([0-9]+\.[0-9]+)",
+        r"GCC: \(GNU\) ([0-9]+\.[0-9]+(\.[0-9]+)?)",
+        # r"gcc ([0-9]+\.[0-9]+(\.[0-9]+)?)",  # does not return correct version number on some packages
```

Comment:
```
        # Make sure the product and version are in the results
        assert (
            product in list_products
        ), f"""{product} not found in {package_name}.
        The checker signature or url may be incorrect."""
>       assert version in list_versions
E       AssertionError: assert '8.4.1' in {'8.3.1'}

test/test_scanner.py:262: AssertionError
```
after scanning a few files, found that `r"gcc ([0-9]+\.[0-9]+(\.[0-9]+)?)"` is not consistent and changed it. 
Also, I was not able to find any appropriate version pattern for debian packages. (I've added a note at top for this)